### PR TITLE
Move CodeGenerator to be accessible by other validation tests

### LINKED ISSUE #2342: Refactor CodeGenerator from val_builtins_test.cpp to be generically usable
I am going to bove this test utility class out to where it can be access by other test files, since I am planning on writting new parameterized tests that would greatly benefit from it.

### PR DESCRIPTION
**Pull request by zoddicus**
Fixes #2342

### CODE REVIEW

File: test/val/val_code_generator.cpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2018 Google LLC.
```
Comment (dj2) [inline test/val/val_code_generator.cpp]: 2019

File: test/val/val_code_generator.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2018 Google LLC.
```
Comment (dj2) [inline test/val/val_code_generator.h]: 2019

File: test/val/CMakeLists.txt [hunk 3]
```diff
@@ -15,6 +15,7 @@
 set(VAL_TEST_COMMON_SRCS
   ${CMAKE_CURRENT_SOURCE_DIR}/../test_fixture.h
   ${CMAKE_CURRENT_SOURCE_DIR}/../unit_spirv.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/val_code_generator.h
```
Comment (dj2) [inline test/val/CMakeLists.txt]: Where is the .cpp file listed in the CMakeLists file?

File: test/val/val_code_generator.h (hunk 2 above)
Comment (zoddicus) [reply test/val/val_code_generator.h]: Done

File: test/val/val_code_generator.cpp (hunk 1 above)
Comment (zoddicus) [reply test/val/val_code_generator.cpp]: Done

File: test/val/CMakeLists.txt (hunk 3 above)
Comment (zoddicus) [reply test/val/CMakeLists.txt]: Done

**Pull request merged**